src/refurbished/rocket.f90

```diff
@@ -356,131 +356,207 @@ subroutine update_trajectory( &
   end subroutine
 
   function rocket( &
-      dt, &
-      tmax, &
-      cp, &
-      mw, &
-      temperature_, &
-      pressure_, &
-      Tflame, &
-      rref, &
-      n, &
-      id, &
-      od, &
-      length, &
-      rhos, &
-      dia, &
-      cf)
+      time_step_length, &
+      simulation_end_time, &
+      heat_capacity_at_constant_pressure, &
+      molecular_weight, &
+      initial_temperature, &
+      initial_pressure, &
+      flame_temperature, &
+      reference_burn_rate, &
+      burn_rate_exponent, &
+      fuel_inner_diameter, &
+      fuel_outer_diameter, &
+      fuel_length, &
+      fuel_density, &
+      nozzle_diameter, &
+      thrust_correction_factor)
     !! this is a basic program of a single stage
     !! rocket motor flowing out of a nozzle, assuming
     !! a thrust coefficient and ignoring the complexities of
     !! what happens to thrust at low pressures, i.e. shock in the nozzle
-    real(dp), intent(in) :: dt
-    real(dp), intent(in) :: tmax
-    real(dp), intent(in) :: cp
-    real(dp), intent(in) :: mw
-    real(dp), intent(in) :: temperature_
-    real(dp), intent(in) :: pressure_
-    real(dp), intent(in) :: Tflame
-    real(dp), intent(in) :: rref
-    real(dp), intent(in) :: n
-    real(dp), intent(in) :: id
-    real(dp), intent(in) :: od
-    real(dp), intent(in) :: length
-    real(dp), intent(in) :: rhos
-    real(dp), intent(in) :: dia
-    real(dp), intent(in) :: cf
+    real(dp), intent(in) :: time_step_length
+    real(dp), intent(in) :: simulation_end_time
+    real(dp), intent(in) :: heat_capacity_at_constant_pressure
+    real(dp), intent(in) :: molecular_weight
+    real(dp), intent(in) :: initial_temperature
+    real(dp), intent(in) :: initial_pressure
+    real(dp), intent(in) :: flame_temperature
+    real(dp), intent(in) :: reference_burn_rate
+    real(dp), intent(in) :: burn_rate_exponent
+    real(dp), intent(in) :: fuel_inner_diameter
```

Comment:
fuel should really be called propellant because it contains both the fuel and oxidizer together

Reply:
in all cases...